lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h

```diff
@@ -1037,8 +1014,45 @@ class SwiftASTContextForExpressions : public SwiftASTContext {
 
   void ModulesDidLoad(ModuleList &module_list);
 
-private:
-  std::unique_ptr<SwiftPersistentExpressionState> m_persistent_state_up;
+  typedef llvm::StringMap<swift::AttributedImport<swift::ImportedModule>>
+      HandLoadedModuleSet;
+
+  // This just adds this module to the list of hand-loaded modules, it doesn't
+  // actually load it.
+  void AddHandLoadedModule(
+      ConstString module_name,
+      swift::AttributedImport<swift::ImportedModule> attributed_import) {
+    m_hand_loaded_modules.insert_or_assign(module_name.GetStringRef(),
+                                           attributed_import);
+  }
+
+  /// Retrieves the modules that need to be implicitly imported in a given
+  /// execution scope. This includes the modules imported by both the compile
+  /// unit as well as any imports from previous expression evaluations.
+  bool GetImplicitImports(
+      SymbolContext &sc, lldb::ProcessSP process_sp,
+      llvm::SmallVectorImpl<swift::AttributedImport<swift::ImportedModule>>
+          &modules,
+      Status &error);
+
+  // FIXME: the correct thing to do would be to get the modules by calling
+  // CompilerInstance::getImplicitImportInfo, instead of loading these
+  // modules manually. However, we currently don't have  access to a
+  // CompilerInstance, which is why this function is needed.
+  void LoadImplicitModules(lldb::TargetSP target, lldb::ProcessSP process,
+                           ExecutionContextScope &exe_scope);
+  /// Cache the user's imports from a SourceFile in a given execution scope such
+  /// that they are carried over into future expression evaluations.
+  bool CacheUserImports(lldb::ProcessSP process_sp,
+                        swift::SourceFile &source_file, Status &error);
+
+protected:
+  /// These are the names of modules that we have loaded by hand into
+  /// the Contexts we make for parsing.
+  HandLoadedModuleSet m_hand_loaded_modules;
+
+  /// This returns the list of hand-loaded modules.
+  HandLoadedModuleSet GetHandLoadedModules() { return m_hand_loaded_modules; }
```

Comment:
Could this be returned as a const reference instead of by copy?

```suggestion
  const HandLoadedModuleSet &GetHandLoadedModules() { return m_hand_loaded_modules; }
```